### PR TITLE
Splash Damage Patch

### PR DESCRIPTION
**Pull request by MrDuck557**
Before:
Hail failed to damage navanax even after visually hitting it
navanax
Scatter splash damage fails to damage antumbra properly, and the base damage is cut down by armor

https://user-images.githubusercontent.com/48808663/145219294-94bded66-6b1c-4e26-8c93-cc313e8e2cec.mp4

After:
This actually works
navanax
Antumbra dies properly, taking the 40 dmg splash

https://user-images.githubusercontent.com/48808663/145219541-f1d98241-af0e-43e3-8029-b11b14f3cdd6.mp4

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by MrDuck557** [comment]
I know that checks failed, this is the log
bm github thing

File: core/src/mindustry/entities/Damage.java [hunk 1]
```diff
@@ -334,14 +334,14 @@ public static void damage(Team team, float x, float y, float radius, float damag
...
-            if(entity.team == team || !entity.within(x, y, radius) || (entity.isFlying() && !air) || (entity.isGrounded() && !ground)){
+            if(entity.team == team || !entity.within(x, y, radius + entity.type.hitSize/2) || (entity.isFlying() && !air) || (entity.isGrounded() && !ground)){
                 return;
             }
 
             entity.apply(effect, duration);
         };
 
-        rect.setSize(radius * 2).setCenter(x, y);
+        rect.setSize((radius + 100) * 2).setCenter(x, y);
```
Comment (Anuken) [inline core/src/mindustry/entities/Damage.java:344]: All queries take hitbox size into account when checking collisions; this shouldn't be necessary. Same goes for the other occurrence of this.

Comment (MrDuck557) [reply core/src/mindustry/entities/Damage.java:344]: fixing

**Comment by LeoSko** [comment]
This patch makes sense to me, but it also significantly affects current balance. If it gets merged, can we also adjust healths/damages accordingly?

**Comment by Anuken** [comment]
Yeah, now that I look at, it:

> Antumbra dies properly, taking the 40 dmg splash

...has very serious balance consequences. This is essentially a massive buff for flak against already-weak air T4s and T5s, which is not acceptable.

**Comment by Anuken** [comment]
> can we also adjust healths/damages accordingly?

This isn't a viable strategy - there are far too many combinations of turrets/units, and basic value changes like "buff antumbra health to compensate" would indirectly nerf non-splash-damage weapons.

**Comment by MrDuck557** [comment]
time to apply a different solution

**Comment by LeoSko** [comment]
> This isn't a viable strategy - there are far too many combinations of turrets/units, and basic value changes like "buff antumbra health to compensate" would indirectly nerf non-splash-damage weapons.

Of course it doesn't. Balance is mainly done not via characteristics but abilities and tactics (~~hopefully we will get an obvious scheme what beats what some time huh, if any?~~). But it also doesn't mean we shouldn't do anything or not adjust balance at all after that particular buff.

**Comment by Anuken** [comment]
> Balance is mainly done not via characteristics but abilities and tactics

Up to this point, I've addressed the overwhelming majority of unit/turret balancing problems by just changing some numbers - most problems *can* be solved this way.

I think the best solution here is to 1) do nothing (it's not a real problem, in my opinion) or 2) apply minor visual tweaks so that e.g. hails *do* deal damage to the navanax, but it's floored at 1, or some other tiny number that cannot possibly affect balance.

**Comment by Anuken** [comment]
Won't be merging this regardless due to the balance implications.

**Pull request closed without merging**